certification/internal/policy/operator/deployable_by_olm_test.go

```diff
@@ -0,0 +1,71 @@
+package operator
+
+import (
+	"os"
+	"path/filepath"
+
+	log "github.com/sirupsen/logrus"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/internal/utils/migration"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/cli"
+)
+
+var _ = Describe("DeployableByOLMCheck", func() {
+	var (
+		deployableByOLMCheck DeployableByOlmCheck
+		fakeOpenshiftEngine  cli.OpenshiftEngine
+	)
+	const (
+		metadataDir        = "metadata"
+		annotationFilename = "annotations.yaml"
+	)
+
+	BeforeEach(func() {
+		nsOpenshiftReport := cli.OpenshiftReport{
+			Stdout: `{
+				"apiVersion": "v1",
+				"kind": "namespace",
+				"name": "operator-ns"
+			  }
+			  `,
+			Stderr: "",
+		}
+		fakeOpenshiftEngine = FakeOpenshiftEngine{
+			OpenshiftReport: nsOpenshiftReport,
+		}
+		deployableByOLMCheck = *NewDeployableByOlmCheck(&fakeOpenshiftEngine)
+
+		var err error
+		annotations := `annotations:
+  operators.operatorframework.io.bundle.package.v1: testPackage
+  operators.operatorframework.io.bundle.channel.default.v1: testChannel
+`
+		// set env var for index image
+		os.Setenv("PFLT_INDEXIMAGE", "test_indeximage")
+		os.Setenv("PFLT_ARTIFACTS", metadataDir)
+
+		// mock bundle directory
+		err = os.Mkdir(metadataDir, 0755)
+		Expect(err).ToNot(HaveOccurred())
+		err = os.WriteFile(filepath.Join(metadataDir, annotationFilename), []byte(annotations), 0644)
+		Expect(err).ToNot(HaveOccurred())
+	})
+	Describe("When deploying an operator using OLM", func() {
+		Context("When CSV has been created successfully", func() {
+			It("Should pass Validate", func() {
+				log.Info("************ " + os.Getenv("PFLT_INDEXIMAGE"))
+				ok, err := deployableByOLMCheck.Validate(migration.ImageToImageReference("dummy/image"))
```

Comment:
The migration struct doesn't exist anymore. Everything should be using certification.ImageReference now

Reply:
I'll upload a new patch in which I fixed this.